api/grpc/pb/api.proto

```diff
@@ -59,6 +59,52 @@ message MsgError {
     string error = 1;
 }
 
+message MsgErrorV2 {
+    ErrorCategory category = 1;
+    ErrorCode code = 2;
+    string message = 3;
+    oneof addInfo {
+        ErrV2InfoResourceNotFound ErrV2InfoResourceNotFound = 4;
```

Comment:
What do you mean by this `oneof` ?

Reply:
`oneof` directive in proto implies, the data in this field can be one of the mentioned types.

In the additional info field, the data type will change depending upon the error code. To implement this in proto, i found two options: `any` and `oneof`.

In case of `any`, the data should be marshal every-time you write data to this and unmarshal   every time when it should be read. This also means, there could be error (and associated error handling) when using this option. Also, `any` allows the user to put data of any type in this field.

But in case of `oneof`, one can specify what are the types that are allowed in this field. Also, the concrete data can be retrieved by "type extraction" (with no error handling required); and compitable data types can just be assigned to it.

Hence i chose to use `oneof` over `any`.

Reply:
But, here only one type is mentioned, in that case `oneof` is not required right?

Reply:
Yes, in this PR I have added just one field as a place holder. Other fields will be added in the following PRs. For examples, you can find 2 more types being added in the PR that follows this: #148 and #149.